src/main/java/net/dries007/tfc/common/entities/predator/Predator.java

```diff
@@ -45,24 +46,37 @@ public static AttributeSupplier.Builder createAttributes()
     public static final EntityDataAccessor<Boolean> DATA_SLEEPING = SynchedEntityData.defineId(Predator.class, EntityDataSerializers.BOOLEAN);
 
     private static int ATTACK_ANIMATION_LENGTH;
+    @Nullable
+    public Vec3 location;
+    @Nullable
+    public Vec3 prevLocation;
+    public float walkProgress = 0f;
+    public static int WALK_ANIMATION_LENGTH;
 
     public final boolean diurnal;
     private int attackAnimationRemainingTicks = 0;
+    public final boolean isMale= random.nextBoolean();
```

Comment:
bad equals sign

Reply:
also you have to serialize this value or you will change the gender of the animal each time its loaded

Reply:
Okay, so I still don't really understand what's wrong with having an equals sign, but I assume it's a style thing.

Also wasn't sure what it meant to serialize a value, and while I still don't actually know what it means, I now have a working knowledge that allowed me to fix the problem.

Also, while testing to see if I'd fixed the problem, I discovered that predators despawned when you left the chunks since they extend Mob, which I fixed by just setting persistence required. Idk if there's a better way to do that, but it's at least a band-aid